### PR TITLE
tbi(cap-config-writer): Support getting additional info about CAP in one call

### PR DESCRIPTION
**Pull request by IainSAP**
https://github.com/SAP/open-ux-tools/issues/1689

Since there is already work being done to read CAP project files we should allow this information to be returned to avoid additional expensive calls (reading and parsing files from disk). 

- Adds overloaded `checkCdsUi5PluginEnabled` and adds additional optional parameter `moreInfo`
- Adds new type for new return object
- Adds tests to cover

### CODE REVIEW

File: packages/cap-config-writer/src/cap-config/index.ts [hunk 1]
```diff
@@ -37,12 +38,32 @@ export async function enableCdsUi5Plugin(basePath: string, fs?: Editor): Promise
...
+ * Overloaded function that returns detailed CAP plugin info.
  *
  * @param basePath - root path of the CAP project, where package.json is located
  * @param [fs] - optional: the memfs editor instance
- * @returns true: cds-plugin-ui5 and all prerequisites are fulfilled; false: cds-plugin-ui5 is not enabled or not all prerequisites are fulfilled
+ * @param [moreInfo] if true return an object specifying detailed info about the cds and workspace state
+ * @returns true: cds-plugin-ui5 and all prerequisites are fulfilled; false: cds-plugin-ui5 is not enabled or not all prerequisites are fulfilled or {@link: CdsUi5PluginInfo} with additional info
  */
-export async function checkCdsUi5PluginEnabled(basePath: string, fs?: Editor): Promise<boolean> {
+export async function checkCdsUi5PluginEnabled(
```
Comment (IainSAP) [inline packages/cap-config-writer/src/cap-config/index.ts:48]: Not sure this is necessary...will check.

File: packages/cap-config-writer/package.json [hunk 2]
```diff
@@ -19,6 +19,7 @@
         "lint": "eslint . --ext .ts",
         "lint:fix": "eslint . --ext .ts --fix",
         "test": "jest --ci --forceExit --detectOpenHandles --colors",
+        "test-u": "jest --ci --forceExit --detectOpenHandles --colors -u",
```
Comment (longieirl) [inline packages/cap-config-writer/package.json:22]: Is this required?

Comment (IainSAP) [reply packages/cap-config-writer/package.json:22]: Nope, thx @longieirl

**Review by longieirl — APPROVED** [review]
Changes look good.

**Review by tobiasqueck — APPROVED** [review]
* changeset good
* test coverage excellent
* new functionality is clear and properly implemented
* I was surprised of the remove of the `i18next` dependency, I assume it was a cleanup of an unnecessary dependency independent of the change
* didn't test manually but I trust the jest tests

**Pull request merged**